packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/react/basic-sign-in.md

```diff
@@ -0,0 +1,202 @@
+  const onError = (err) => {
+    console.log('Sign in error:', err);
+  };
+
+  if (!authState) {
+    return <div>Loading ... </div>;
+  }
+
+  return authState.isAuthenticated ?
+    <Redirect to={{ pathname: '/' }}/> :
+    <OktaSignInWidget config={config} onSuccess={onSuccess} onError={onError}/>;
+};
+export default Login;
+```
+
+#### Callback route
+
+The [Okta React SDK](https://github.com/okta/okta-react) provides the [LoginCallback](https://github.com/okta/okta-react#logincallback) component for the callback route. It handles token parsing, token storage, and redirects the user to the `/` path. If a `SecureRoute` triggered the redirect, then the callback is directed to the secured route. See how the callback route component is called from the route definition file (`src/App.jsx`) in the [Connect the routes](#connect-the-routes) section.
+
+### Connect the routes
+
+Update `src/App.jsx` to include your project components and routes. The `Security` component controls the authentication flows, so it requires your OpenID Connect configuration. By default, `@okta/okta-react` redirects to Okta's sign-in page when the user isn't authenticated. In this example, `onAuthRequired` is overridden to redirect to the custom sign-in route instead.
+
+> **Note:** This example uses the `react-router-dom` version 5.x module for the `useHistory` and `Route` objects.
+
+```js
+import React from 'react';
+import { Route, useHistory } from 'react-router-dom';
+import { Security, SecureRoute, LoginCallback } from '@okta/okta-react';
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
+import Home from './Home';
+import Login from './Login';
+import Protected from './Protected';
+import Profile from './Profile';
+import config from './config';
+
+import logo from './logo.svg';
+import './App.css';
+
+const oktaAuth = new OktaAuth(config.oidc);
+
+const App = () => {
+  const history = useHistory();
+
+  const customAuthHandler = () => {
+    history.push('/login');
+  };
+
+  const restoreOriginalUri = async (_oktaAuth, originalUri) => {
+    history.replace(toRelativeUrl(originalUri || '', window.location.origin));
+  };
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>My Okta-React app</p>
+        <img src={logo} className="App-logo" alt="logo" />
+      <Security
+        oktaAuth={oktaAuth}
+        onAuthRequired={customAuthHandler}
+        restoreOriginalUri={restoreOriginalUri}
+      >
+        <Route path="/" exact component={Home} />
+        <SecureRoute path="/protected" component={Protected} />
+        <Route path="/login" render={() => <Login />} />
+        <Route path="/login/callback" component={LoginCallback} />
+    </Security>
+    </header>
+    </div>
+  );
+};
+
+export default App;
+
+```
+
+### Start your app
+
+To run and test your app, execute:
+
+```js
+npm start
+```
+
+Open a browser and navigate to your app URL. For example: [http://localhost:3000](http://localhost:3000)
```

Comment:
```suggestion
Open a browser and navigate to your app at [http://localhost:3000](http://localhost:3000).
```

Reply:
I don't know if this step is necessary since Create React App will open your default browser for you.